### PR TITLE
Sign up password field criteria

### PR DESCRIPTION
**Pull request by Shruti-Apte**
### **Notion :**
- https://www.notion.so/saeloun/Signin-Signup-review-mobile-fc999aac28374cb0921d200822fdf718?pvs=4
- https://www.notion.so/saeloun/Mobile-Fix-the-spacing-b-w-logo-and-text-on-sign-in-page-d8dda2a6aa3543c7931dd5d517199f33?pvs=4
### **What :**
- Password field criteria added and conditions added for error field.
- Sign In mobile view page spacing fixed between logo and welcome back text.
### **Why :**
- User should know the password criteria beforehand
- We were showing same error for password and confirm password field (bad UI/UX)  
- Too much space between logo and welcome back text.
### **Preview :**
Sign In page before:
Screenshot 2023-05-11 at 1 04 03 PM

Sign In page After:
Screenshot 2023-05-11 at 12 37 50 PM

Signup page:
Desktop:
Screenshot 2023-05-11 at 12 50 19 PM

Screenshot 2023-05-11 at 1 05 43 PM

Mobile:
Screenshot 2023-05-11 at 12 38 17 PM

Screenshot 2023-05-11 at 12 49 42 PM

### CODE REVIEW

File: app/javascript/src/components/Authentication/SignUp/SignUpForm.tsx [hunk 1]
```diff
@@ -161,17 +167,15 @@ const SignUpForm = () => {
...
-                      fieldTouched={touched.password}
+                      wrapperClassName="mb-6"
                     />
                   </div>
                   <div className="field">
                     <InputField
                       id="confirm_password"
                       label="Confirm Password"
                       labelClassName="p-0"
+                      marginBottom="mb-2"
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Authentication/SignUp/SignUpForm.tsx:178]: Use classnames and provide the tailwind classes to override the inpux styles

**Review by prasanthchaduvula — CHANGES_REQUESTED** [review]

File: app/javascript/src/components/Authentication/SignUp/SignUpForm.tsx (hunk 1 above)
Comment (Shruti-Apte) [reply app/javascript/src/components/Authentication/SignUp/SignUpForm.tsx:178]: If the new value that we provide to override the default value is smaller than default value, then it won’t work even with the `classnames`
e.g: If the default value is `mb-6` and we want to override it with `mb-2`, then it won’t work.

That’s why I added different `marginBottom`

**Comment by prasanthchaduvula** [comment]
@Shruti-Apte Always the end class names will override the previous ones

**Comment by Shruti-Apte** [comment]
> @Shruti-Apte Always the end class names will override the previous ones

Not in case of smaller values.

**Comment by prasanthchaduvula** [comment]
@Shruti-Apte we can resolve the classNames in another PR

**Review by prasanthchaduvula — APPROVED** [review]

**Pull request merged**